models/articles.models.js

```diff
@@ -6,24 +6,37 @@ exports.selectArticleById = (id) => {
     const queryString = 'SELECT * FROM articles WHERE article_id=$1'
     return db.query(queryString, [articleId])
     .then(({rows}) => {
+        if(rows[0] === undefined){
+            return Promise.reject({status: 404, msg : 'Not Found'})
+        } 
         return rows
     })
 }
 
-exports.selectArticles = () => {
-    const queryString = `SELECT articles.author, 
+exports.selectArticles = ({topic}) => {
+    
+    let topicString = ''
+    if(topic){
+        if(topic !== 'paper' && topic !== 'mitch' && topic !== 'cats' && topic !== 'undefined'){
```

Comment:
Ideally we don't want to hard code these topics as they could expand or change. We want to use the database to get all available topics and then check is it one of them.